From 21897dba67545140e2aad0d08e0779194e6d3c73 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 1 Jul 2025 11:46:38 +0200
Subject: [PATCH 915/954] feat(stm32mp): add stack protector support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add stack protector in strong mode when debug is enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git b5a43827d54ccb82271c27740dd1b37c5338d30f feat(stm32mp): add stack protector support]
Change-Id: I51f24a497cb0a14194e1c3552f48e78fcc7527fc
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/471310
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/common/common.mk                                    | 6 ++++++
 .../stm32mp_stack_protector.c}                              | 0
 plat/st/stm32mp1/platform.mk                                | 4 ----
 3 files changed, 6 insertions(+), 4 deletions(-)
 rename plat/st/{stm32mp1/stm32mp1_stack_protector.c => common/stm32mp_stack_protector.c} (100%)

diff --git a/plat/st/common/common.mk b/plat/st/common/common.mk
index d6805ce81e..f7a5d4217e 100644
--- a/plat/st/common/common.mk
+++ b/plat/st/common/common.mk
@@ -65,6 +65,12 @@ STM32_TF_A_COPIES		:=	2
 # PLAT_PARTITION_MAX_ENTRIES must take care of STM32_TF-A_COPIES and other partitions
 PLAT_PARTITION_MAX_ENTRIES	:=	$(shell echo $$(($(STM32_TF_A_COPIES) + $(STM32_EXTRA_PARTS))))
 
+ifneq ($(ENABLE_STACK_PROTECTOR),0)
+# Enable the stack protector. If set "to strong", this improves
+# debugging at the cost of a size increase.
+PLAT_BL_COMMON_SOURCES		+=	plat/st/common/stm32mp_stack_protector.c
+endif
+
 ifeq (${PSA_FWU_SUPPORT},1)
 # Number of banks of updatable firmware
 NR_OF_FW_BANKS			:=	2
diff --git a/plat/st/stm32mp1/stm32mp1_stack_protector.c b/plat/st/common/stm32mp_stack_protector.c
similarity index 100%
rename from plat/st/stm32mp1/stm32mp1_stack_protector.c
rename to plat/st/common/stm32mp_stack_protector.c
diff --git a/plat/st/stm32mp1/platform.mk b/plat/st/stm32mp1/platform.mk
index 2e48c1f283..d9c4c7c490 100644
--- a/plat/st/stm32mp1/platform.mk
+++ b/plat/st/stm32mp1/platform.mk
@@ -227,10 +227,6 @@ PLAT_BL_COMMON_SOURCES	+=	plat/st/stm32mp1/stm32mp1_private.c
 
 PLAT_BL_COMMON_SOURCES	+=	drivers/st/uart/aarch32/stm32_console.S
 
-ifneq (${ENABLE_STACK_PROTECTOR},0)
-PLAT_BL_COMMON_SOURCES	+=	plat/st/stm32mp1/stm32mp1_stack_protector.c
-endif
-
 PLAT_BL_COMMON_SOURCES	+=	lib/cpus/aarch32/cortex_a7.S
 
 PLAT_BL_COMMON_SOURCES	+=	drivers/arm/tzc/tzc400.c				\
-- 
2.53.0

